Make PerformanceStats rendering more efficient

DougBanksPersonal opened the pull request:
Instead of clearing previous bars and creating new bars each time we render stats graph, try to re-use existing bars.

Also don't re-position the target line every render; it never moves.

hockeychaos commented on CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua:
```diff
@@ -132,10 +132,19 @@ end
-  local isVisible = (GameSettings.PerformanceStatsVisible and localPlayer ~= nil)
-  if isVisible then 
+  local shouldBeVisible = StatsUtils.PerformanceStatsShouldBeVisible()
+  print("\n\nUpdatePerformanceStatsVisibility: shouldBeVisible = ", shouldBeVisible);
```
lets remove this print

DougBanksPersonal replied:
Done.

hockeychaos commented on CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua:
```diff
@@ -132,10 +132,19 @@ end
     masterFrame.Parent = CoreGuiService.RobloxGui
+    
+    -- Refresh, the contents are stale.
+    statsViewer:Refresh();
```
mostly a style thing, but don't need the semicolon

DougBanksPersonal replied:
Done

hockeychaos commented on CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua:
```diff
@@ -132,10 +132,19 @@ end
+    statsViewer:Refresh();
+    for i, buttonType in ipairs(StatsUtils.AllStatTypes) do
+        local button = statsButtonsByType[buttonType]
+        button:Refresh();
```
same there here wrt semicolon

DougBanksPersonal replied:
Done

hockeychaos commented on CoreScriptsRoot/Modules/Stats/StatsAggregator.lua:
```diff
@@ -61,6 +61,11 @@ end
   -- On a regular heartbeat, wake up and read the latest
   -- value into circular buffer.
+  -- Don't bother if we're already listening.
+  if (self._listneing == true) then 
```
typo

DougBanksPersonal replied:
Done

hockeychaos commented on CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua:
```diff
@@ -176,7 +176,11 @@ function StatsAnnotatedGraphClass:_recursiveGetOrderOfMagnitude(estimate, target
 
-function StatsAnnotatedGraphClass:_updateValue()
+function StatsAnnotatedGraphClass:UpdateValuesAndRender()
+  if not StatsUtils.PerformanceStatsShouldBeVisible() then 
```
would it make sense to register/desregister listeners instead of doing if checks like this? Would be more performant to only make these calls if we need to

DougBanksPersonal replied:
Will look into it.  Stay tuned.

DougBanksPersonal commented:
OK I believe I have addressed your points.

Thanks
Doug

On Mon, Oct 10, 2016 at 3:13 PM, Ben Tkacheff <notifications@github.com>
wrote:

> *@hockeychaos* requested changes on this pull request.
> ------------------------------
>
> In CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>
> :
>
> > @@ -132,10 +132,19 @@ end
>  function UpdatePerformanceStatsVisibility()
>    local localPlayer = PlayersService.LocalPlayer
>
> -  local isVisible = (GameSettings.PerformanceStatsVisible and localPlayer ~= nil)
> -  if isVisible then
> +  local shouldBeVisible = StatsUtils.PerformanceStatsShouldBeVisible()
> +  print("\n\nUpdatePerformanceStatsVisibility: shouldBeVisible = ", shouldBeVisible);
>
> lets remove this print
> ------------------------------
>
> In CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>
> :
>
> >      masterFrame.Visible = true
>      masterFrame.Parent = CoreGuiService.RobloxGui
> +
> +    -- Refresh, the contents are stale.
> +    statsViewer:Refresh();
>
> mostly a style thing, but don't need the semicolon
> ------------------------------
>
> In CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>
> :
>
> >      masterFrame.Visible = true
>      masterFrame.Parent = CoreGuiService.RobloxGui
> +
> +    -- Refresh, the contents are stale.
> +    statsViewer:Refresh();
> +    for i, buttonType in ipairs(StatsUtils.AllStatTypes) do
> +        local button = statsButtonsByType[buttonType]
> +        button:Refresh();
>
> same there here wrt semicolon
> ------------------------------
>
> In CoreScriptsRoot/Modules/Stats/StatsAggregator.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>
> :
>
> > @@ -61,6 +61,11 @@ end
>  function StatsAggregatorClass:StartListening()
>    -- On a regular heartbeat, wake up and read the latest
>    -- value into circular buffer.
> +  -- Don't bother if we're already listening.
> +  if (self._listneing == true) then
>
> typo
> ------------------------------
>
> In CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>
> :
>
> > @@ -176,7 +176,11 @@ function StatsAnnotatedGraphClass:_recursiveGetOrderOfMagnitude(estimate, target
>    return self:_recursiveGetOrderOfMagnitude(estimate*10.0, target)
>  end
>
> -function StatsAnnotatedGraphClass:_updateValue()
> +function StatsAnnotatedGraphClass:UpdateValuesAndRender()
> +  if not StatsUtils.PerformanceStatsShouldBeVisible() then
>
> would it make sense to register/desregister listeners instead of doing if
> checks like this? Would be more performant to only make these calls if we
> need to
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3581774>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHHDJfq1PV36O3dgnAurXK_yYufIGxjnks5qyriggaJpZM4KS-gr>
> .
>

hockeychaos commented on CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua:
```diff
@@ -176,7 +208,11 @@ function StatsAnnotatedGraphClass:_recursiveGetOrderOfMagnitude(estimate, target
 end
 
-function StatsAnnotatedGraphClass:_updateValue()
+local counter = 0
```
whats up with this counter?

hockeychaos commented on CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua:
```diff
@@ -155,13 +155,45 @@ function StatsAnnotatedGraphClass:SetStatsAggregator(aggregator)
+    self:_maybeStartListening()
+    self:_updateValuesAndRender()
+  else
+    self:_maybeStopListening()
```
the maybe seems ambiguous. To the person calling this function we will definitely stop listening when we call this, just the function itself might not have to do anything as it already is not listening. Can we call it just stopListening and startListening since the maybe seems like an implementation detail?

DougBanksPersonal commented:
Done, please take a look.

On Mon, Oct 10, 2016 at 5:40 PM, Ben Tkacheff <notifications@github.com>
wrote:

> *@hockeychaos* commented on this pull request.
> ------------------------------
>
> In CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3596455>
> :
>
> > @@ -176,7 +208,11 @@ function StatsAnnotatedGraphClass:_recursiveGetOrderOfMagnitude(estimate, target
>    return self:_recursiveGetOrderOfMagnitude(estimate*10.0, target)
>  end
>
> -function StatsAnnotatedGraphClass:_updateValue()
> +local counter = 0
>
> whats up with this counter?
> ------------------------------
>
> In CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3596455>
> :
>
> >    end
>
> -  self:_updateValue()
> +  self._listenerId = self._aggregator:AddListener(function()
> +      self:_updateValuesAndRender()
> +  end)
> +end
> +
> +
> +
> +function StatsAnnotatedGraphClass:OnVisibilityChanged()
> +  if StatsUtils.PerformanceStatsShouldBeVisible() then
> +    self:_maybeStartListening()
> +    self:_updateValuesAndRender()
> +  else
> +    self:_maybeStopListening()
>
> the maybe seems ambiguous. To the person calling this function we will
> definitely stop listening when we call this, just the function itself might
> not have to do anything as it already is not listening. Can we call it just
> stopListening and startListening since the maybe seems like an
> implementation detail?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ROBLOX/Core-Scripts/pull/577#pullrequestreview-3596455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHHDJc8hvP4cNC8s55t-OTBngvANkcLAks5qytrtgaJpZM4KS-gr>
> .
>

DougBanksPersonal commented:
OK.
Can I resolve the Jira ticket (
https://jira.roblox.com/browse/CLISTUDIO-10755)?
If so, what CL number?

Thanks
Doug

On Tue, Oct 11, 2016 at 10:37 AM, Ben Tkacheff <notifications@github.com>
wrote:

> Merged #577 <https://github.com/ROBLOX/Core-Scripts/pull/577>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ROBLOX/Core-Scripts/pull/577#event-819847403>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHHDJaVVNSkALruyxKFqa9ycJtdVZe_Zks5qy8lUgaJpZM4KS-gr>
> .
>
